node/src/components/linear_chain_sync.rs

```diff
@@ -202,83 +397,71 @@ where
                             peer
                         );
                         // NOTE: Signal misbehaving validator to networking layer.
+                        // NOTE: Cannot call `self.ban_peer` with `peer` value b/c it's fixed for
+                        // `KeyFingerprint` type and we're abstract in what
+                        // peer type is.
                         return self.handle_event(
                             effect_builder,
                             rng,
-                            Event::GetBlockResult(block_hash, None),
+                            Event::GetBlockHashResult(block_hash, None),
                         );
                     }
                     trace!(%block_hash, "Downloaded linear chain block.");
-                    self.reset_peers();
-                    self.new_block(*block.clone());
-                    let curr_height = block.height();
-                    // We instantiate with `highest_block_seen=0`, start downloading with the
-                    // highest block and then download its ancestors. It should
-                    // be updated only once at the start.
-                    if curr_height > self.highest_block_seen {
-                        self.highest_block_seen = curr_height;
-                    }
+                    self.add_block(*block.clone());
+
                     if block.is_genesis_child() {
```

Comment:
Can we just check `block.height() == 0` instead?

Reply:
It's the same thing and `is_genesis_child()` tells the reader what's going on, instead of comparing it to _some_ number.